client/my-sites/stats/stats-date-control/stats-date-control-picker.tsx

```diff
@@ -7,18 +7,67 @@ import DateControlPickerShortcuts from './stats-date-control-picker-shortcuts';
 import { DateControlPickerProps } from './types';
 
 const DateControlPicker = ( { slug, queryParams }: DateControlPickerProps ) => {
+	const redirectToQueryDate = ( daysToSubtract: number ) => {
+		const startDate = calculateQueryDate( daysToSubtract );
+		const query = qs.stringify( Object.assign( {}, queryParams, { startDate } ), {
+			addQueryPrefix: true,
+		} );
+		const period = 'day'; // TODO: adjust this as needed
+		const url = `/stats/${ period }/${ slug }`;
+		const href = `${ url }${ query }`;
+
+		page( href );
+	};
+
 	// shortcut list will come from props
 	const shortcutList = [
 		{
-			id: 'test',
-			label: 'Test shortcut',
-			onClick: () => {
-				// eslint-disable-next-line no-console
-				console.log( 'clicking' );
-			},
+			id: 'today',
+			label: 'Today',
+			onClick: () => redirectToQueryDate( 0 ),
+		},
+		{
+			id: 'yesterday',
+			label: 'Yesterday',
+			onClick: () => redirectToQueryDate( 1 ),
+		},
+		{
+			id: 'last-7-days',
+			label: 'Last 7 Days',
+			onClick: () => redirectToQueryDate( 7 ),
+		},
+		{
+			id: 'last-30-days',
+			label: 'Last 30 Days',
+			onClick: () => redirectToQueryDate( 30 ),
+		},
+		{
+			id: 'last-year',
+			label: 'Last Year',
+			onClick: () => redirectToQueryDate( 365 ),
+		},
+		{
+			id: 'all-time',
+			label: 'All Time',
+			onClick: () => {}, // eslint-disable-line @typescript-eslint/no-empty-function
+			// This is empty because I don't yet fully understand what 'all time' will show
+		},
+		{
+			id: 'custom-range',
+			label: 'Custom Range',
+			onClick: () => {}, // eslint-disable-line @typescript-eslint/no-empty-function
+			// Keep this empty, since the custom range is handled by the DateControlPickerDate component.
```

Comment:
I'm not sure "Custom Range" should be something the user selects. I was thinking this would be set by the control when the current range didn't match any of the shortcuts. Or maybe it just auto-focuses the first input field when selected? We'll probably need more input from the team on this one.